### PR TITLE
Clean makefiles, move GPG signing to release.sh

### PR DESCRIPTION
**Pull request by igor-hnizdo**
I have cleaned up the makefiles. I have also moved the GPG signing to the individual releases and out of existing Makefile.

Also I read the GPG key out of GPG instead of hard-coding it.

The negative is that now we require quite a lot of keys to set up... especially the macos now needs tons of keys.

I will try to document them all somewhere.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,37 +1,7 @@
...
-
-BUILD_DIR    := build
-TARGETS      := $(foreach platform,$(PLATFORMS),$(BUILD_DIR)/$(NAME)-$(platform))
-
-all: $(TARGETS)
-
-$(BUILD_DIR):
-	mkdir -p $(BUILD_DIR)
-
-$(BUILD_DIR)/$(NAME)-linux-arm64: $(BUILD_DIR)
```
Comment (igor-hnizdo) [inline Makefile:18]: I deleted all those, I don't think it's useful for anything really

File: release/Makefile [hunk 2]
```diff
@@ -14,13 +14,15 @@ windows:
...
+	cp macos/build/*.pkg macos/build/*.pkg.asc installers
+	cp windows/build/trezor-bridge*.exe windows/build/trezor-bridge*.exe.asc installers
 	cp linux/build/*.deb linux/build/*.rpm installers
-
-gpg:
-	gpg -u 54067D8BBF00554181B5AB8F26A3A56662F0E7E2 --armor --detach-sig installers/trezor-bridge*.exe
-	gpg -u 54067D8BBF00554181B5AB8F26A3A56662F0E7E2 --armor --detach-sig installers/*.pkg
+	rm -rf binaries
+	mkdir binaries
+	cp macos/build/trezord binaries/trezord-darwin-universal
```
Comment (igor-hnizdo) [inline release/Makefile:22]: I have added new folder "binaries" for all the built binaries. (Different than the installers.)

File: release/macos/release.sh [hunk 3]
```diff
@@ -157,3 +157,12 @@ if [ -r $NOT_KEY_ID_F ]; then
...
     rm $NOT_JSON
 fi
+
+GPG_PRIVKEY=/release/gpg-privkey.asc
+if [ -r $GPG_PRIVKEY ]; then
+    export GPG_TTY=$(tty)
+    export LC_ALL=C.UTF-8
+    gpg --import /release/privkey.asc
+    GPGSIGNKEY=$(gpg --list-keys --with-colons | grep '^pub' | cut -d ":" -f 5)
+    gpg -u $GPGSIGNKEY --armor --detach-sig /release/build/*.pkg
```
Comment (igor-hnizdo) [inline release/macos/release.sh:167]: btw I am not exactly sure if we really need to GPG sign the macOS and windows releases when they are already signed and macOS even notarized, but well, we did it already so I just moved it here

Comment (tsusanka) [reply release/macos/release.sh:167]: Yeah I agree, but we do that for Suite as well, so let's keep it for now.

File: release/linux/release.sh [hunk 4]
```diff
@@ -21,6 +20,7 @@ if [ -r $GPG_PRIVKEY ]; then
     export LC_ALL=C.UTF-8
     gpg --import /release/privkey.asc
     GPG_SIGN=gpg
+    GPGSIGNKEY=$(gpg --list-keys --with-colons | grep '^pub' | cut -d ":" -f 5)
```
[the remark this answers is not in the document]
Comment (tsusanka) [reply release/linux/release.sh:23]: Just ccing @prusnak if he knows something better

File: Makefile [hunk 5]
```diff
@@ -1,37 +1,7 @@
...
-
-$(BUILD_DIR)/$(NAME)-linux-x64: $(BUILD_DIR)
-	CC=gcc CGO_ENABLED=1 GOOS=linux GOARCH=amd64 CGO_CFLAGS="$(LINUX_CFLAGS)" go build $(GOFLAGS) -o $(BUILD_DIR)/$(NAME)-linux-x64
-
-$(BUILD_DIR)/$(NAME)-mac-x64: $(BUILD_DIR)
-	CC=x86_64-apple-darwin14-clang CGO_ENABLED=1 GOOS=darwin GOARCH=amd64 CGO_CFLAGS="$(MAC_CFLAGS)" go build $(GOFLAGS) -o $(BUILD_DIR)/$(NAME)-mac-x64
-
-$(BUILD_DIR)/$(NAME)-win-x64: $(BUILD_DIR)
-	CC=x86_64-w64-mingw32-gcc CGO_ENABLED=1 GOOS=windows GOARCH=amd64 CGO_CFLAGS="$(WIN_CFLAGS)" go build $(GOFLAGS) -o $(BUILD_DIR)/$(NAME)-win-x64
+all: native
```
Comment (tsusanka) [inline Makefile:1]: Let's drop this altogether?

**Review by tsusanka — APPROVED** [review]

File: Makefile (hunk 5 above)
Comment (igor-hnizdo) [reply Makefile:1]: OK

**Review by tsusanka — COMMENTED** [review]
I get an error for `make` in release/ because I do not have the certs. Could you please change it that it does not sign when there are no such files?

Last nits, we I will merge then.

File: Makefile [hunk 6]
```diff
@@ -1,37 +1,7 @@
...
 native:
 	CGO_ENABLED=1 go build $(GOFLAGS)
 
-check:
-	file $(TARGETS)
-
-clean:
-	rm -f $(TARGETS)
+release:
+	cd release && make
```
[the remark this answers is not in the document]
Comment (igor-hnizdo) [reply Makefile:7]: oops

**Comment by igor-hnizdo** [comment]
Fixed

**Comment by igor-hnizdo** [comment]
OK I will see if it's still doing that. I thought I fixed that by Makefile changes

**Comment by igor-hnizdo** [comment]
I have tested it and it works without the keys

**Comment by tsusanka** [comment]
I am getting this on 073154d5dea5d40b7c35f1b9b63dcfb8ca92e711:

```
+ rcodesign sign --p12-file /release/certs/app.p12 --p12-password-file /release/certs/app.passphrase --code-signature-flags runtime /release/build/trezord
Error: I/O error: No such file or directory (os error 2)
make[1]: *** [.package] Error 1
make: *** [macos] Error 2
```

**Comment by igor-hnizdo** [comment]
ahh it's not the GPG stuff, but something else. OK will check that

**Comment by tsusanka** [comment]
Oh yeah, sorry it is the certs.

**Pull request merged**